### PR TITLE
workflows: rework, trivy & publish

### PR DESCRIPTION
**Pull request by tkatila**
Fixes: #882

### CODE REVIEW

File: .github/workflows/devel.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Devel
+on:
+  schedule:
+    - cron: '0 4 * * *'
```
Comment (mythi) [inline .github/workflows/devel.yaml:4]: we have used e2e "nightly" trigger to follow e2e is "alive" but AFAICS with this modification we end up doing a nightly push too?

File: .github/workflows/lib-publish.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          scan-type: image
+          image-ref: intel/${{ matrix.image }}:${{ inputs.image_tag }}
+          exit-code: 1
+      - name: Login
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_PASS }}
+      - name: Push
```
Comment (mythi) [inline .github/workflows/lib-publish.yaml:58]: we may want `make test-image-base-layer` (and drop `push:` from `Makefile`?)

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  tag_fix:
+    name: Prepare image tag
+    runs-on: ubuntu-22.04
+    outputs:
+      fixed_tag: ${{ steps.fix.outputs.tag }}
+    env:
+      TAGNAME: ${{ github.ref_name }}
+    steps:
+    - id: fix
+      # remove first character (v) and keep the next 12
```
Comment (mythi) [inline .github/workflows/release.yaml:20]: what's 12 for?

File: .trivyignore [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+AVD-DS-0002
+
+# Privileged containers share namespaces with the host system and do not offer any security.
+# They should be used exclusively for system containers that require high  # privileges.
+# initcontainers require privileged access
+AVD-KSV-0017
+
+# Sharing the host’s network namespace permits processes in the pod to communicate with
+# processes bound to the host’s loopback adapter.
+# sgx requires hostNetwork: true
```
Comment (mythi) [inline .trivyignore:14]:
```suggestion
# sgx single-node demo deployment uses hostNetwork: true to be able to use the default PCCS URL from containers
```

File: .github/workflows/devel.yaml (hunk 1 above)
Comment (tkatila) [reply .github/workflows/devel.yaml:4]: That's correct. I can re-create the e2e workflow and set it to run periodically, and remove the schedule from devel.

File: .github/workflows/release.yaml (hunk 3 above)
Comment (tkatila) [reply .github/workflows/release.yaml:20]: Reason is silly. I thought that one had to set both start and end indices, but apparently it works fine with just `${TAGNAME:1}`

File: .gitignore [hunk 5]
```diff
@@ -34,3 +34,7 @@ _work
 
 *.tgz
 charts/operator/crds
```
Comment (mythi) [inline .gitignore:36]: these seem to be some leftovers from our helm work. while we're at it, maybe drop?

File: .github/workflows/lib-validate.yaml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    - name: Install terrascan
+      run: |
+        curl -sL "$(curl -sL https://api.github.com/repos/tenable/terrascan/releases/latest | grep -o -E "https://.+?_Linux_x86_64.tar.gz")" > terrascan.tar.gz
+        tar -xf terrascan.tar.gz terrascan && rm terrascan.tar.gz
+        install terrascan /usr/local/bin && rm terrascan
+    - name: Run Terrascan
+      run: make terrascan
```
Comment (mythi) [inline .github/workflows/lib-validate.yaml:84]: we could just drop this in favor of trivy

**Review by mythi — COMMENTED** [review]
amazing work, thanks! two small suggestions and maybe then squash for merging

could add "Fixes: #882"

File: .github/workflows/lib-validate.yaml (hunk 6 above)
Comment (tkatila) [reply .github/workflows/lib-validate.yaml]: dropped

File: .gitignore (hunk 5 above)
Comment (tkatila) [reply .gitignore]: sure, removed

**Review by mythi — APPROVED** [review]

**Pull request merged**